features/nanostack/nanostack-interface/Nanostack.cpp

```diff
@@ -648,6 +648,18 @@ nsapi_size_or_error_t Nanostack::do_sendto(void *handle, const ns_address_t *add
      * */
     if (retcode == NS_EWOULDBLOCK) {
         ret = NSAPI_ERROR_WOULD_BLOCK;
+    } else if (retcode == -1) {
```

Comment:
Feels like this is getting to the point it should be a switch statement for the error cases, although I guess you still need an if for the `< 0`.

Reply:
Yes, that's why I stayed with it if-else statements.

Reply:
Just a tad wary that the compiler will end up doing all the comparisons in the order specified. Compilers are generally good at spotting how to break a switch up into ranges and figuring out when to do a branch table, and when to do binary chop, but I'm not sure how good they are at optimising an if-else chain on 1 variable.

At least make the non-error case the first, if going down this route, for speed.

It might be time to use a non-conventional `if-else-switch` control construct:

    if (retcode >= 0) {
       ret = retcode;
    } else switch (retcode) {
       case -1:
           ret = NSAPI_ERROR_PARAMETER;
           break;
       ...
       default:
           tr_error();
           ret = NSAPI_ERROR_DEVICE_ERROR;
           break;
    }

(Would look neater if Mbed OS style didn't insist on the extra switch indent).

Reply:
Done